src/ArmTemplates/Extractor/EntityExtractors/ServiceApisProductsExtractor.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clients.Abstractions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Constants;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Exceptions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Extensions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Abstractions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.EntityExtractors.Abstractions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.EntityExtractors
+{
+    public class ServiceApisProductsExtractor : TemplateGeneratorBase, IServiceApisProductsExtractor
+    {
+        readonly ILogger<ServiceApisProductsExtractor> logger;
+        
+        readonly IServiceApiProductsApiClient serviceApiProductsApiClient;
+        readonly IServiceApisApiClient serviceApisApiClient;
+
+        public ServiceApisProductsExtractor(
+            ILogger<ServiceApisProductsExtractor> logger,
+            IServiceApiProductsApiClient productApisApiClient,
+            IServiceApisApiClient serviceApiClient)
+        {
+            this.logger = logger;
+            
+            this.serviceApiProductsApiClient = productApisApiClient;
+            this.serviceApisApiClient = serviceApiClient;
+        }
+
+        public async Task<Template> GenerateServiceApisProductsARMTemplateAsync(string singleApiName, List<string> multipleApiNames, ExtractorParameters extractorParameters)
```

Comment:
I think we need to break this method up into different ones

```suggestion
        public async Task<Template> GenerateServiceApisProductsARMTemplateAsync(List<string> multipleApiNames, ExtractorParameters extractorParameters)
        public async Task<Template> GenerateServiceApisProductsARMTemplateAsync(string singleApiName, ExtractorParameters extractorParameters)
```

Reply:
Didn't want to expose another API so from `ExtractorExecutor` perspective nothing is changing. But did change the method significantly internally.